src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const modifyedObjects = [];
+  const typeValueAdd = 'addProperties';
+  const typeValueRemove = 'removeProperties';
+  const typeValueClear = 'clear';
+
+  for (const actionObject of actions) {
+    switch (actionObject.type) {
+      case typeValueAdd:
+        const kyes = Object.keys(actionObject.extraData);
+        const values = Object.values(actionObject.extraData);
+
+        for (let i = 0; i < kyes.length; i++) {
+          copyState[kyes[i]] = values[i];
+        }
+        pushValue(modifyedObjects, { ...copyState });
+        break;
+      case typeValueRemove:
```

Comment:
```suggestion
        break;
        
      case typeValueRemove:
```
Add empty lines between cases to improve readability